### PR TITLE
♻️ director-v2 no longer captures all env vars

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

- ⚰️ removed some unused env vars
- ♻️ capture only the minimum required env vars to run

## Dev Checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

## DevOps Checklist

- ⚠️ `director-v2` no longer uses `.env` file

### CODE REVIEW

File: services/docker-compose.yml [hunk 1]
```diff
@@ -202,38 +202,64 @@ services:
...
-      - DIRECTOR_HOST=${DIRECTOR_HOST:-director}
-      - DIRECTOR_PORT=${DIRECTOR_PORT:-8080}
       - DIRECTOR_SELF_SIGNED_SSL_FILENAME=${DIRECTOR_SELF_SIGNED_SSL_FILENAME}
       - DIRECTOR_SELF_SIGNED_SSL_SECRET_ID=${DIRECTOR_SELF_SIGNED_SSL_SECRET_ID}
       - DIRECTOR_SELF_SIGNED_SSL_SECRET_NAME=${DIRECTOR_SELF_SIGNED_SSL_SECRET_NAME}
       - DIRECTOR_SERVICES_CUSTOM_CONSTRAINTS=${DIRECTOR_SERVICES_CUSTOM_CONSTRAINTS}
-      - DIRECTOR_V2_DEV_FEATURES_ENABLED=${DIRECTOR_V2_DEV_FEATURES_ENABLED}
-      - DYNAMIC_SIDECAR_IMAGE=${DOCKER_REGISTRY:-itisfoundation}/dynamic-sidecar:${DOCKER_IMAGE_TAG:-latest}
-      - EXTRA_HOSTS_SUFFIX=${EXTRA_HOSTS_SUFFIX:-undefined}
+      - TRACING_THRIFT_COMPACT_ENDPOINT=${TRACING_THRIFT_COMPACT_ENDPOINT}
```
Comment (sanderegg) [inline services/docker-compose.yml:209]: why do you need that entry?
As far as I can see, this is only used in the webserver?

File: services/docker-compose.yml [hunk 2]
```diff
@@ -202,38 +202,64 @@ services:
...
-      - MONITORING_ENABLED=${MONITORING_ENABLED:-True}
-      - SIMCORE_SERVICES_NETWORK_NAME=interactive_services_subnet
-      - TRACING_THRIFT_COMPACT_ENDPOINT=${TRACING_THRIFT_COMPACT_ENDPOINT}
+      - DYNAMIC_SIDECAR_ENABLE_VOLUME_LIMITS=${DYNAMIC_SIDECAR_ENABLE_VOLUME_LIMITS}
+      - TRAEFIK_SIMCORE_ZONE=${TRAEFIK_SIMCORE_ZONE:-internal_simcore_stack}
       - DYNAMIC_SIDECAR_PROMETHEUS_SERVICE_LABELS=${DYNAMIC_SIDECAR_PROMETHEUS_SERVICE_LABELS}
       - DYNAMIC_SIDECAR_PROMETHEUS_MONITORING_NETWORKS=${DYNAMIC_SIDECAR_PROMETHEUS_MONITORING_NETWORKS}
-    env_file:
-      - ../.env
+      - DYNAMIC_SIDECAR_DOCKER_NODE_SAVES_LOCK_TIMEOUT_S=${DYNAMIC_SIDECAR_DOCKER_NODE_SAVES_LOCK_TIMEOUT_S}
```
Comment (sanderegg) [inline services/docker-compose.yml:238]: was this not removed?

File: services/docker-compose.yml [hunk 3]
```diff
@@ -202,38 +202,64 @@ services:
...
       - DIRECTOR_SELF_SIGNED_SSL_SECRET_NAME=${DIRECTOR_SELF_SIGNED_SSL_SECRET_NAME}
       - DIRECTOR_SERVICES_CUSTOM_CONSTRAINTS=${DIRECTOR_SERVICES_CUSTOM_CONSTRAINTS}
-      - DIRECTOR_V2_DEV_FEATURES_ENABLED=${DIRECTOR_V2_DEV_FEATURES_ENABLED}
-      - DYNAMIC_SIDECAR_IMAGE=${DOCKER_REGISTRY:-itisfoundation}/dynamic-sidecar:${DOCKER_IMAGE_TAG:-latest}
-      - EXTRA_HOSTS_SUFFIX=${EXTRA_HOSTS_SUFFIX:-undefined}
+      - TRACING_THRIFT_COMPACT_ENDPOINT=${TRACING_THRIFT_COMPACT_ENDPOINT}
       - LOG_LEVEL=${LOG_LEVEL:-WARNING}
       - LOG_FORMAT_LOCAL_DEV_ENABLED=${LOG_FORMAT_LOCAL_DEV_ENABLED}
+      - DIRECTOR_V2_DEV_FEATURES_ENABLED=${DIRECTOR_V2_DEV_FEATURES_ENABLED}
+      - SWARM_STACK_NAME=${SWARM_STACK_NAME:-simcore}
```
Comment (sanderegg) [inline services/docker-compose.yml:213]: remove default values

File: services/docker-compose.yml [hunk 4]
```diff
@@ -202,38 +202,64 @@ services:
...
+      - SIMCORE_SERVICES_NETWORK_NAME=interactive_services_subnet
```
Comment (sanderegg) [inline services/docker-compose.yml:214]: remove hard-coded value and move to .env-devel

File: services/docker-compose.yml [hunk 5]
```diff
@@ -202,38 +202,64 @@ services:
...
+      - MONITORING_ENABLED=${MONITORING_ENABLED:-True}
```
Comment (sanderegg) [inline services/docker-compose.yml:215]: remove default

File: services/docker-compose.yml [hunk 6]
```diff
@@ -202,38 +202,64 @@ services:
...
+      - STORAGE_HOST=${STORAGE_HOST}
+      - STORAGE_PORT=${STORAGE_PORT}
+      - CATALOG_HOST=${CATALOG_HOST:-catalog}
```
Comment (sanderegg) [inline services/docker-compose.yml:218]: default

File: services/docker-compose.yml [hunk 7]
```diff
@@ -202,38 +202,64 @@ services:
...
+      - CATALOG_PORT=${CATALOG_PORT:-8000}
+      - DIRECTOR_HOST=${DIRECTOR_HOST:-director}
+      - DIRECTOR_PORT=${DIRECTOR_PORT:-8080}
```
Comment (sanderegg) [inline services/docker-compose.yml:221]: defaults

File: services/docker-compose.yml [hunk 8]
```diff
@@ -202,38 +202,64 @@ services:
...
+      - DYNAMIC_SIDECAR_LOG_LEVEL=${DYNAMIC_SIDECAR_LOG_LEVEL}
+      - DYNAMIC_SIDECAR_IMAGE=${DOCKER_REGISTRY:-itisfoundation}/dynamic-sidecar:${DOCKER_IMAGE_TAG:-latest}
```
Comment (sanderegg) [inline services/docker-compose.yml:223]: defaults

**Review by sanderegg — COMMENTED** [review]
cool, please check my comments and remove all the defaults as defined in the best practices.
defaults should move to .env-devel.
Also there are some ENV that I think are useless

File: services/docker-compose.yml [hunk 9]
```diff
@@ -202,38 +202,64 @@ services:
...
       - LOG_FORMAT_LOCAL_DEV_ENABLED=${LOG_FORMAT_LOCAL_DEV_ENABLED}
+      - DIRECTOR_V2_DEV_FEATURES_ENABLED=${DIRECTOR_V2_DEV_FEATURES_ENABLED}
+      - SWARM_STACK_NAME=${SWARM_STACK_NAME:-simcore}
+      - SIMCORE_SERVICES_NETWORK_NAME=interactive_services_subnet
+      - MONITORING_ENABLED=${MONITORING_ENABLED:-True}
+      - STORAGE_HOST=${STORAGE_HOST}
+      - STORAGE_PORT=${STORAGE_PORT}
+      - CATALOG_HOST=${CATALOG_HOST:-catalog}
+      - CATALOG_PORT=${CATALOG_PORT:-8000}
+      - DIRECTOR_HOST=${DIRECTOR_HOST:-director}
```
Comment (YuryHrytsuk) [inline services/docker-compose.yml:220]: defaults

File: services/docker-compose.yml [hunk 10]
```diff
@@ -202,38 +202,64 @@ services:
...
+      - POSTGRES_PORT=${POSTGRES_PORT}
+      - POSTGRES_USER=${POSTGRES_USER}
+      - REDIS_HOST=${REDIS_HOST}
+      - REDIS_PORT=${REDIS_PORT}
+      - RABBIT_HOST=${RABBIT_HOST}
+      - RABBIT_PASSWORD=${RABBIT_PASSWORD}
+      - RABBIT_PORT=${RABBIT_PORT}
+      - RABBIT_SECURE=${RABBIT_SECURE}
+      - RABBIT_USER=${RABBIT_USER}
+      - COMPUTATIONAL_BACKEND_DEFAULT_CLUSTER_URL=${COMPUTATIONAL_BACKEND_DEFAULT_CLUSTER_URL:-tcp://dask-scheduler:8786}
```
Comment (YuryHrytsuk) [inline services/docker-compose.yml:251]: defaults

**Comment by YuryHrytsuk** [comment]
Thank you. Amazing initiative :rocket:

File: services/docker-compose.yml (hunk 1 above)
Comment (GitHK) [reply services/docker-compose.yml]: Good point dropped it

File: services/docker-compose.yml [hunk 11]
```diff
@@ -202,38 +202,62 @@ services:
...
+      - COMPUTATIONAL_BACKEND_DEFAULT_FILE_LINK_TYPE=${COMPUTATIONAL_BACKEND_DEFAULT_FILE_LINK_TYPE}
+      - COMPUTATIONAL_BACKEND_ON_DEMAND_CLUSTERS_FILE_LINK_TYPE=${COMPUTATIONAL_BACKEND_ON_DEMAND_CLUSTERS_FILE_LINK_TYPE}
+      - REGISTRY_USER=${REGISTRY_USER}
+      - REGISTRY_PW=${REGISTRY_PW}
+      - REGISTRY_URL=${REGISTRY_URL}
+      - REGISTRY_SSL=${REGISTRY_SSL}
+      - REGISTRY_AUTH=${REGISTRY_AUTH}
+      - REGISTRY_PATH=${REGISTRY_PATH}
+      - RESOURCE_USAGE_TRACKER_HOST=${RESOURCE_USAGE_TRACKER_HOST}
+      - DIRECTOR_V2_SERVICES_CUSTOM_CONSTRAINTS=${DIRECTOR_V2_SERVICES_CUSTOM_CONSTRAINTS}
```
Comment (sanderegg) [inline services/docker-compose.yml:260]: better, can we have them alphabetically sorted also?

**Review by sanderegg — APPROVED** [review]
very cool, thanks a bunch!

**Review by YuryHrytsuk — APPROVED** [review]
Great job. Amazing contribution. Thank you!

File: .env-devel [hunk 12]
```diff
@@ -58,6 +58,8 @@ DYNAMIC_SIDECAR_PROMETHEUS_MONITORING_NETWORKS=[]
 
 FUNCTION_SERVICES_AUTHORS='{"UN": {"name": "Unknown", "email": "unknown@osparc.io", "affiliation": "unknown"}}'
 
+MONITORING_ENABLED=True
```
Comment (pcrespov) [inline .env-devel:61]: AFAIK you should also add this in all configs

This is a part of the instructions by @YuryHrytsuk 
image

I could not see at least `MONITORING_ENABLED=True` there

Comment (GitHK) [reply .env-devel]: Yes, removed it already

**Pull request merged**